src/main/kotlin/org/springdoc/openapi/gradle/plugin/OpenApiGeneratorTask.kt

```diff
@@ -63,19 +64,28 @@ open class OpenApiGeneratorTask : DefaultTask() {
 
     private fun generateApiDocs(url: String, fileName: String) {
         try {
+            val isYaml = url.toLowerCase().matches(Regex(".+[./]yaml(/.+)*"))
             await ignoreException ConnectException::class withPollInterval Durations.ONE_SECOND atMost Duration.of(
                 waitTimeInSeconds.get().toLong(),
                 SECONDS
             ) until {
-                val statusCode = khttp.get(url).statusCode
+                val url = URL(url)
+                val connection: HttpURLConnection = url.openConnection() as HttpURLConnection
```

Comment:
Shadowing of local variables should generally be avoided
```suggestion
                val connection: HttpURLConnection = URL(url).openConnection() as HttpURLConnection
```